src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java

```diff
@@ -28,20 +30,114 @@
  */
 public class Operation {
 
+    private final Enforcer e;
+
+    public Operation(Enforcer e) {
+        this.e = e;
+    }
+
+    /**
+     * hasAuthorization determines whether the user currently logged on has a permission of the object.
+     *
+     * @param act the action (usually means the operation that the user performs on the resource).
+     * @return whether the current login user has the permission.
+     */
+    public boolean hasAuthorization(String act) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.enforce(subject.getPrincipal().toString(), act);
+    }
+
     /**
-     * hasAuthorization determines whether the current login user has a permission.
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
```

Comment:
Trailing s?

Reply:
Maybe trailing s is wrong.I will take your advice.